File: frontend/CodeInTasks/src/app/web-api/models/task/task-update-model.ts
import RunnerType from "../../enums/runner-type";
import TaskCategory from "../../enums/task-category";

export default class TaskUpdateModel {
    public title: string = null!;
    public description: string = null!;

    public category: TaskCategory = null!;
    public runner: RunnerType = null!;

    public baseRepositoryUrl: string = null!;

    public testRepositoryUrl: string = null!;
    public testRepositoryAuthPassword: string = null!;
}

File: backend/Shared/src/CodeInTasks.WebApi.Models/Task/TaskUpdateModel.cs
namespace CodeInTasks.WebApi.Models.Task
{
    public class TaskUpdateModel
    {
        [Required]
        [MinLength(DomainConstants.TaskModel_Title_MinLength)]
        [MaxLength(DomainConstants.TaskModel_Title_MaxLength)]
        public string Title { get; set; }

        [Required]
        [MinLength(DomainConstants.TaskModel_Description_MinLength)]
        [MaxLength(DomainConstants.TaskModel_Description_MaxLength)]
        public string Description { get; set; }

        public TaskCategory Category { get; set; }

        public RunnerType Runner { get; set; }

        [Required]
        [MinLength(DomainConstants.RepositoryUrl_MinLength)]
        [MaxLength(DomainConstants.RepositoryUrl_MaxLength)]
        public string BaseRepositoryUrl { get; set; }

        [Required]
        [MinLength(DomainConstants.RepositoryUrl_MinLength)]
        [MaxLength(DomainConstants.RepositoryUrl_MaxLength)]
        public string TestRepositoryUrl { get; set; }

        [Required]
        [MinLength(DomainConstants.RepositoryAuthPassword_MinLength)]
        [MaxLength(DomainConstants.RepositoryAuthPassword_MaxLength)]
        public string TestRepositoryAuthPassword { get; set; }
    }
}
